x-pack/plugins/observability_solution/dataset_quality/public/hooks/use_redirect_link.ts

```diff
@@ -43,29 +47,66 @@ export const useRedirectLink = ({
   const logsExplorerLocator =
     share.url.locators.get<SingleDatasetLocatorParams>(SINGLE_DATASET_LOCATOR_ID);
 
-  const config = logsExplorerLocator
-    ? buildLogsExplorerConfig({
-        locator: logsExplorerLocator,
-        dataStreamStat,
-        query,
-        from,
-        to,
-        breakdownField,
-      })
-    : buildDiscoverConfig({
-        locatorClient: share.url.locators,
-        dataStreamStat,
-        query,
-        from,
-        to,
-        breakdownField,
-      });
-
-  return {
-    ...config.routerLinkProps,
-    navigate: config.navigate,
-    isLogsExplorerAvailable: !!logsExplorerLocator,
-  };
+  const { sendTelemetry } = useRedirectLinkTelemetry({
+    rawName: dataStreamStat.rawName,
+    isLogsExplorer: !!logsExplorerLocator,
+    telemetry,
+    query,
+  });
+
+  return useMemo<{
+    linkProps: RouterLinkProps;
+    navigate: () => void;
+    isLogsExplorerAvailable: boolean;
+  }>(() => {
+    const config = logsExplorerLocator
+      ? buildLogsExplorerConfig({
+          locator: logsExplorerLocator,
+          dataStreamStat,
+          query,
+          from,
+          to,
+          breakdownField,
+        })
+      : buildDiscoverConfig({
+          locatorClient: share.url.locators,
+          dataStreamStat,
+          query,
+          from,
+          to,
+          breakdownField,
+        });
+
+    const onClickWithTelemetry = (event: Parameters<RouterLinkProps['onClick']>[0]) => {
+      sendTelemetry();
+      if (config.routerLinkProps.onClick) {
+        config.routerLinkProps.onClick(event);
+      }
+    };
+
+    const navigateWithTelemetry = () => {
+      sendTelemetry();
+      config.navigate();
+    };
+
+    return {
+      linkProps: {
```

Comment:
`linkProps` which are applied on DOM, needed to be separated in order to avoid the warnings:

```
Warning: React does not recognize the `isLogsExplorerAvailable` prop on a DOM element. ...

Warning: Invalid value for prop `navigate` on <a> tag. Either remove it from the element ...
```